### PR TITLE
Add 'starter app' facet to exchange

### PR DESCRIPTION
**Pull request by proteusvacuum**
@amsagoff what do you think of this? Not too sure about the language, but I think this makes sense.
http://manage.dimagi.com/default.asp?167132#938749
# Admin-only option on publish page

screen shot 2015-05-12 at 12 39 32 pm

There are 2 options on the exchange listing:
# option 1 filter button at the top

screen shot 2015-05-12 at 12 37 44 pm
# option 2 new facet on the side

I thought this was a bit weird, as there would only be one option here.
screen shot 2015-05-12 at 12 38 07 pm

This PR is for option 1, but wouldn't be hard to have option 2 instead.

I also added the "Starter Application" badge to each application tile if they are starter apps. 

Enchantress @gcapalbo 
FYI @czue

### CODE REVIEW

File: corehq/pillows/mappings/domain_mapping.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-DOMAIN_INDEX = "hqdomains_20150403_1448"
-DOMAIN_MAPPING={'_meta': {'comment': 'Yedi modified on 3/20/2014',
+DOMAIN_INDEX = "hqdomains_201505012_1117"
+DOMAIN_MAPPING={'_meta': {'comment': 'Farid modified on 5/12/2015',
```
Comment (dimagimon) [inline corehq/pillows/mappings/domain_mapping.py]: E225 missing whitespace around operator

**Comment by gcapalbo** [comment]
@proteusvacuum This looks like it's good to merge from me.  Might be good for @czue to just double-check the elasticsearch part. Will also see if @amsagoff has any comments.

**Comment by proteusvacuum** [comment]
Discussion here: http://manage.dimagi.com/default.asp?167132#939820

Seems like option 1 is the way to go, so if everyone is good with it, this PR is ready.

**Comment by gcapalbo** [comment]
@czue / @snopoke can one of you just double-check the elasticsearch part? I've never made changes that touch es - all that needs to be done here is just update domain_mapping.py, and not change any pillows or anything else?

**Comment by czue** [comment]
looks good

**Pull request merged**